core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -136,8 +136,13 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite {
     when(mockExternalBlockStoreClient.getHostLocalDirs(any(), any(), any(), any()))
       .thenAnswer { invocation =>
         import scala.jdk.CollectionConverters._
-        invocation.getArgument[CompletableFuture[java.util.Map[String, Array[String]]]](3)
-          .complete(hostLocalDirs.asJava)
+        if (hostLocalDirs != null) {
```

Comment:
Nit: I would prefer avoiding unnecessary negate operators so:

```
if (hostLocalDirs == null) {
  invocation.getArgument[CompletableFuture[java.util.Map[String, Array[String]]]](3)
    .completeExceptionally(new RuntimeException("force fail"))
} else {
  invocation.getArgument[CompletableFuture[java.util.Map[String, Array[String]]]](3)
    .complete(hostLocalDirs.asJava)
}
```

Reply:
Sure, updated. Would like to know why is the preference? Can you please share? Thanks.

Reply:
Readability and cognitive load.